tests/canary_upgrade_test.go

```diff
@@ -199,12 +176,26 @@ var _ = Describe("[sig-operator]virt-handler canary upgrade", Serial, decorators
 		Expect(err).ToNot(HaveOccurred())
 
 		updateTimeout := time.Duration(canaryTestNodeTimeout * nodesToUpdate)
-		Eventually(func() bool {
+		Eventually(func(g Gomega) {
 			expectedEventsLock.Lock()
 			defer expectedEventsLock.Unlock()
-			return len(expectedEvents) == 0
-		}, updateTimeout*time.Second, 1*time.Second).Should(BeTrue(), fmt.Sprintf("events %v were expected but did not occur", expectedEvents))
-
-		Expect(isVirtHandlerUpdated(getVirtHandler())).To(BeTrue())
+			g.Expect(expectedEvents).To(BeEmpty())
+		}, updateTimeout*time.Second, 1*time.Second).Should(Succeed(), fmt.Sprintf("events %v were expected but did not occur", expectedEvents))
+
+		Eventually(func(g Gomega) {
```

Comment:
Shouldn't this be `Expect` as after the canary it should be all done?

Reply:
I think that assumption was the source of the previous flakiness tbh and resulted in the update to status being missed. I'm using `Eventually` here to be defensive and allow for the initial checks to fail.

Reply:
Can you elaborate? I am not sure I follow.